src/before-all.ts

```diff
@@ -0,0 +1,50 @@
+import "reflect-metadata"
+import { TestOptionsInput, parseTestOptions, isTestOptionsInput } from "./options"
+import { configuration } from "./configure"
+
+export type BeforeAllFunction = (instance: any) => void
+
+export function getBeforeAlls(target: Object): BeforeAllFunction[] {
+    const beforeAlls = Reflect.getMetadata("test-decorator:beforeAll", target)
+    if (typeof beforeAlls !== "undefined") { return beforeAlls }
+    const newBeforAllContainer: BeforeAllFunction[] = []
+    Reflect.defineMetadata("test-decorator:beforeAll", newBeforAllContainer, target)
+    return newBeforAllContainer
+}
+
+export type TestDecorator =
```

Comment:
I think this should be called `BeforeAllDecorator` as it is the the signature of the beforeAll decorator and not the test decorator. As it has the same signature as the `TestDecorator` and perhaps others will also have, we should move this into a file `types.ts` or similar and call it `SuiteMethodDecorator` or similar.

Reply:
thx! nice catch